### PR TITLE
global: command-steps pattern and cli groups

### PR DESCRIPTION
**Pull request by ppanero**
**Requires** https://github.com/inveniosoftware/cookiecutter-invenio-rdm/pull/124

**Changes**

- Refactor commands into its own module and split by type
- Refactor cli and split by group to improve readability
- Abstracted process calling and response types (e.g. `ExecResult` from docker and compose, others from `subprocess`, etc.) now are encapsulated in a `ProcessResponse`. Same is done for calls.
- Most of the commands, now return a series of steps. Except some like (`ensure_containers_are_running`, `update_statics_and_assets` and others that turned more complex when a step requires the output of the previous or use complex env vars.). Then the CLI `click.commands` are running this step and showing feedback to the user.

*Tests related*
- Split tests per case instead of having several cases in one test
- Minimize imports (this greatly helped to know what and why something was being patched).
- Caveat: Had to skip some tests, for now, tested flows manually and they work.

**Triggers**
- When agreed to merge this, `FIXME`s need to be turned into issues (e.g. turn the above complex operations into steps)

**Reference Documentation**
https://github.com/inveniosoftware/docs-invenio-rdm/pull/85

Closes https://github.com/inveniosoftware/invenio-cli/issues/180

### CODE REVIEW

File: invenio_cli/cli/assets.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        force=force,
+        flask_env='production' if production else 'development'
+    )
+
+
+@assets.command()
+@click.argument('path', type=click.Path(exists=True))
+@pass_cli_config
+def install(cli_config, path):
+    """Install and link a React module."""
```
Comment (lnielsen) [inline invenio_cli/cli/assets.py:43]: Comment: Not sure, but perhaps this should mention it's fro development?

Comment (ppanero) [reply invenio_cli/cli/assets.py]: Done, specified for all assets and watch commands.

File: invenio_cli/cli/assets.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          `invenio-cli containers build`.
+    """
+
+
+@assets.command()
+@click.option('--force', '-f', default=False, is_flag=True,
+              help='Force the full recreation the assets and statics.')
+@click.option(
+    '--production/--development', '-p/-d', default=True, is_flag=True,
+    help='Production mode copies files. Development mode symlinks files.'
```
Comment (fenekku) [inline invenio_cli/cli/assets.py:32]: Let's indicate in the help message that `--production` is the default option.

**Comment by dfdan** [comment]
The --no-services flag is a great addition.

Would it be possible to add (commented out) external service config options to invenio.cfg to help people use this?

**Pull request merged**